tests/platform_tests/fwutil/conftest.py

```diff
@@ -81,15 +111,21 @@ def extract_fw_data(fw_pkg_path):
     return fw_data
 
 
-@pytest.fixture(scope='function')
-def random_component(duthost, fw_pkg):
-    chass = list(show_firmware(duthost)["chassis"].keys())[0]
-    components = list(fw_pkg["chassis"].get(chass, {}).get("component", {}).keys())
-    if 'ONIE' in components:
-        components.remove('ONIE')
-    if len(components) == 0:
-        pytest.skip("No suitable components found in config file for platform {}.".format(duthost.facts['platform']))
-    return components[randrange(len(components))]
+@pytest.fixture(scope='function', params=["CPLD", "ONIE", "BIOS", "FPGA"])
+def component(request, duthost, fw_pkg):
+    component_type = request.param
+    if component_type == 'FPGA' and duthost.facts['platform'] not in PLATFORMS_HAVE_FPGA:
+        pytest.skip("Skip the FPGA test for the platform doesn't have FPGA components.")
```

Comment:
@congh-nvidia why do you want to skip FPGA component update?

Reply:
hi @prgeor , the skip is for the platforms that don't have the FPGA component. Any platform that has the FPGA should be add to PLATFORMS_HAVE_FPGA, then it will not be skipped.

Reply:
@prgeor I find that the skip is actually not necessary, the test will be skipped if there is no such component in the firmware.json file. So I removed the skip.